### PR TITLE
feat: support aws kms signer

### PR DESCRIPTION
**Pull request by Nebulis**

### CODE REVIEW

File: src/implementations/utils/wallet.ts [hunk 1]
```diff
@@ -21,33 +36,51 @@ export const getPrivateKey = ({ keyFile, key }: WalletOption): string | undefine
...
+  ...options
+}: WalletOrSignerOption & Partial<NetworkOption> & { progress?: (progress: number) => void }): Promise<
+  Wallet | ConnectedSigner
+> => {
   const provider =
     network === "local"
       ? new providers.JsonRpcProvider()
       : getDefaultProvider(network === "mainnet" ? "homestead" : network); // homestead => aka mainnet
-  if (encryptedWalletPath) {
+  if (isWalletOption(options)) {
```
Comment (cavacado) [inline src/implementations/utils/wallet.ts:51]: would a switch case for the logic branches be better here instead of multiple if-else blocks?

**Review by cavacado — COMMENTED** [review]
lgtm, xcept for 1 logic structure comment

File: src/implementations/utils/wallet.ts [hunk 2]
```diff
@@ -21,33 +36,51 @@ export const getPrivateKey = ({ keyFile, key }: WalletOption): string | undefine
...
+      secretAccessKey, // credentials for your IAM user with KMS access
+      region: options.region,
+      keyId: options.kmsKeyId,
+    };
 
-    if (!privateKey)
-      throw new Error(
-        "No private key found in OA_PRIVATE_KEY, key, key-file, please supply at least one or supply an encrypted wallet path"
-      );
+    const provider = ethers.providers.getDefaultProvider("ropsten");
```
Comment (ChenRen93) [inline src/implementations/utils/wallet.ts:71]: is the awskms only open for ropsten for now ?

Comment (Nebulis) [reply src/implementations/utils/wallet.ts:71]: fuck

Comment (Nebulis) [reply src/implementations/utils/wallet.ts:71]: I forgot to remove this line

**Pull request merged**